examples/dtls-sock/main.c

```diff
@@ -38,9 +31,6 @@ static const shell_command_t shell_commands[] = {
 
 int main(void)
 {
-    /* we need a message queue for the thread running the shell in order to
-     * receive potentially fast incoming networking packets */
-    msg_init_queue(_main_msg_queue, MAIN_QUEUE_SIZE);
```

Comment:
You are still using `gnrc_icmpv6_echo` with `shell_commands` in this example. The corresponding `shell_command` calls `gnrc_netreg_register()` in the main thread, which I [just](https://github.com/RIOT-OS/RIOT/pull/11909#discussion_r309117969) pointed out to you requires the calling thread to have a message queue. 

https://github.com/RIOT-OS/RIOT/blob/dd6a5cde5c84b6deb75cf864e7d70eed6c601461/sys/shell/commands/sc_gnrc_icmpv6_echo.c#L102

 So just removing the queue is the wrong approach.

Reply:
I've split this PR into two and this discussion actually belongs to the new PR, but we can continue here just for this one.

I removed `gnrc_icmpv6_echo` in the latest push. So no `ping6` anymore from the shell for this example. I added it originally for debugging. The example behavior should be the same as the dtls-echo example now. So, in this case, it should be okay, right?

Reply:
> I removed `gnrc_icmpv6_echo` in the latest push. So no `ping6` anymore from the shell for this example. I added it originally for debugging. The example behavior should be the same as the dtls-echo example now. So, in this case, it should be okay, right?

Yepp, under this condition it is correct to remove the queue initialization :-)